modules/secrets-store-aws-secrets-manager.adoc

```diff
@@ -0,0 +1,357 @@
+            privileged: true
+          volumeMounts:
+            - mountPath: "/etc/kubernetes/secrets-store-csi-providers"
+              name: providervol
+            - name: mountpoint-dir
+              mountPath: /var/lib/kubelet/pods
+              mountPropagation: HostToContainer
+      tolerations:
+      - operator: Exists
+      volumes:
+        - name: providervol
+          hostPath:
+            path: "/etc/kubernetes/secrets-store-csi-providers"
+        - name: mountpoint-dir
+          hostPath:
+            path: /var/lib/kubelet/pods
+            type: DirectoryOrCreate
+      nodeSelector:
+        kubernetes.io/os: linux
+----
++
+[NOTE]
+====
+This configuration is modified from the configuration provided in the link:https://github.com/aws/secrets-store-csi-driver-provider-aws#installing-the-aws-provider[AWS documentation] so that it works properly with {product-title}.
+====
+
+.. Grant privileged access to the `csi-secrets-store-provider-aws` service account by running the following command:
++
+[source,terminal]
+----
+$ oc adm policy add-scc-to-user privileged -z csi-secrets-store-provider-aws -n openshift-cluster-csi-drivers
+----
+
+.. Create the provider resources by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f aws-provider.yaml
+----
+
+. Grant permission to allow the service account to read the AWS secret object:
+
+.. Create a directory to contain the credentials request by running the following command:
++
+[source,terminal]
+----
+$ mkdir credentialsrequest-dir-aws
+----
+
+.. Create a YAML file with the following configuration for the credentials request:
++
+.Example `credentialsrequest.yaml` file
+[source,yaml]
+----
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: aws-provider-test
+  namespace: openshift-cloud-credential-operator
+spec:
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: AWSProviderSpec
+    statementEntries:
+    - action:
+      - "secretsmanager:GetSecretValue"
+      - "secretsmanager:DescribeSecret"
+      effect: Allow
+      resource: "arn:*:secretsmanager:*:*:secret:testSecret-??????"
+  secretRef:
+    name: aws-creds
+    namespace: test-namespace
+  serviceAccountNames:
+  - aws-provider
+----
++
+// TODO `resource` value: should we say something about this? I changed it from "secret:xxiaSecret" to "secret:testSecret". What are the "??????" supposed to be?
+
+.. Retrieve the OIDC provider by running the following command:
++
+[source,terminal]
+----
+$ oc get --raw=/.well-known/openid-configuration | jq -r '.issuer'
+----
++
+// TODO: Can this be rewritten to use jsonpath or something else instead of jq? (To avoid making users install jq)
++
+.Example output
+[source,terminal]
+----
+https://<name>-oidc.s3.<aws_region>.amazonaws.com
+----
+Copy the `<oidc_provider_arn>` from the output to use in the next step. For example, `<name>-oidc.s3.<aws_region>.amazonaws.com`.
```

Comment:
As previous comment, in `"issuer":"https://xxxxxxxx"`, the whole `xxxxxxxx` after the `https://` is the OIDC provider name, so it does not make sense to write `<name>` as you type.
And the output does not contain the OIDC provider `arn`, it only contains the OIDC provider name.
And the OIDC provider name does not necessarily to be the format of `xxx-s3.xxx.amazonaws.com`. In ROSA STS cluster, the gotten OIDC name will appear like "d3gtsnipped.cloudfront.net/22nvlqvbbib5l8lj9snipped" (including a slash).
```suggestion
{"issuer":"https://xxxxxxxx",...}
----
Copy the OIDC provider name `xxxxxxxx` from the output to use in the next step.
```

Reply:
Okay I went with this:

```
.Example output
[source,terminal]
----
https://<oidc_provider_name>
----
Copy the OIDC provider name `<oidc_provider_name>` from the output to use in the next step.
```

And then updated the below command to use `<oidc_provider_name>`. Does that work?

```
$ ccoctl aws create-iam-roles \
    --name my-role --region=<aws_region> \
    --credentials-requests-dir=credentialsrequest-dir-aws \
    --identity-provider-arn <oidc_provider_name> --output-dir=credrequests-ccoctl-output
```